fluent.runtime/fluent/runtime/fallback.py

```
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+
+import codecs
+import os
+import six
+
+
+class FluentLocalization(object):
+    """
+    Generic API for Fluent applications.
+
+    This handles language fallback, bundle creation and string localization.
+    It uses the given resource loader to load and parse Fluent data.
+    """
+    def __init__(
+        self, locales, resource_ids, resource_loader,
+        use_isolating=False,
+        bundle_class=None, functions=None,
+    ):
+        self.locales = locales
+        self.resource_ids = resource_ids
+        self.resource_loader = resource_loader
+        self.use_isolating = use_isolating
+        if bundle_class is None:
+            from fluent.runtime import FluentBundle
+            self.bundle_class = FluentBundle
+        else:
+            self.bundle_class = bundle_class
+        self.functions = functions
+        self._bundle_cache = []
+        self._bundle_it = self._iterate_bundles()
+
+    def format_value(self, msg_id, args=None):
+        for bundle in self._bundles():
+            if not bundle.has_message(msg_id):
+                continue
+            msg = bundle.get_message(msg_id)
+            if not msg.value:
+                continue
+            val, errors = bundle.format_pattern(msg.value, args)
+            return val
```

Maybe return the errors, too?